File: TestSample/php/src/Person.php
<?php
declare(strict_types=1);

namespace HelloWorld;

abstract class Person
{
    public function __construct(
        public readonly string $name,
        public readonly string $language
    ) {}

    abstract public function speak(): string;

    public function hello(): string
    {
        return sprintf("[%s] %s: %s", $this->language, $this->name, $this->speak());
    }
}

File: TestSample/csharp/src/Person.cs
namespace HelloWorld;

public abstract class Person
{
    public string Name { get; }
    public string Language { get; }

    protected Person(string name, string language)
    {
        Name = name;
        Language = language;
    }

    public abstract string Speak();

    public string Hello() => $"[{Language}] {Name}: {Speak()}";
}
